app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -333,6 +335,19 @@ private void showListPanel(int type) {
         mDialogFragment = dialogFragment;
     }
 
+    private void dismissAllMenus() {
+        if (menu != null) {
+            menu.dismiss();
+        }
+        BrowserFragment browserFragment = getVisibleBrowserFragment();
+        if (browserFragment != null) {
+            browserFragment.dismissWebContextMenu();
+        }
+        if (mDialogFragment != null) {
```

Comment:
@idiotmax do you think that
```
if (mDialogFragment != null && safeForFragmentTransactions) {
    mDialogFragment.dismiss();
}
```
will be better?

Reply:
@idiotmax pointed out that due to this is called in onNewIntent(), it will never be safe. it should be easier to simply allow state loss.